pkg/datapath/loader/netlink.go

```diff
@@ -692,3 +695,62 @@ func renameDevice(from, to string) error {
 
 	return nil
 }
+
+// DeviceHasTCProgramLoaded checks whether a given device has tc filter/qdisc progs attached.
+func DeviceHasTCProgramLoaded(ifc string) (bool, error) {
```

Comment:
What if this function takes the direction as an argument and then the caller can basically do:
```
loaded, err = loader.DeviceHasTCProgramLoaded(ep.HostInterface(), INGRESS)
...
if ep.RequiresEgressProg {
    loaded, err = loader.DeviceHasTCProgramLoaded(ep.HostInterface(), EGRESS)
    ...
}
...
```
I think this would reduce the complexity of this function and make it easy to know when we need to check for egress progs. WDYT?

Reply:
I am not keen on the idea. I think leaving this up to the caller to decide is less than ideal. IMHO best to change the call to take in the ep and let it decide whether it should check for egress based on the ep nature. 
Would this make better sense?